doc/kernel/threads/lifecycle.rst

```diff
@@ -191,6 +205,43 @@ The following code has the same effect as the code segment above.
                     my_entry_point, NULL, NULL, NULL,
                     MY_PRIORITY, 0, K_NO_WAIT);
 
+User Mode Constraints
+---------------------
+
+This section only applies if :option:`CONFIG_USERSPACE` is enabled, and a user
+thread tries to create a new thread. The :c:func:`k_thread_create()` API is
+still used, but there are additional constraints:
+
+* The child thread object must be in an uninitailized state, i.e. it is
+  not currently running.
+
+* The child's stack object must also be uninitialized.
+
+* The stack size parameter passed in must be equal to or less than the
+  bounds of the stack object when it was declared.
+
+* The :c:macro:`K_USER` option must be used, as user threads can only create
+  other user threads.
+
+* The :c:macro:`K_ESSENTIAL` option must not be used, user threads may not be
+  considered essential threads.
+
+* The priority of the child thread must be a valid priority value, and equal to
+  or worse than the parent thread.
```

Comment:
I rather strongly disagree here.
This leads to confusion because lower integral priority values have better priority in Zephyr. Priority -1 has "higher" priority than 1.
It is much less confusing to use "better" or "worse" than use "higher" and "lower" since the semantics are flipped

Reply:
Well, and "worse" sounds as if there's something wrong with it (the priority) ;-).

Reply:
I did a quick grep through the docs for "priority" and all the other Zephyr discussions of threads use "higher" and "lower" (e.g., http://docs.zephyrproject.org/kernel/threads/scheduling.html ).   Also, header files such as kernel.h  talk about "thread of the same or higher priority".

Reply:
@dbkinder in my opinion the rest of the documentation should be changed to use "better" or "worse" but for the scope of this patch I'll be consistent with the rest of it.